src/snowflake/snowpark/_internal/compiler/large_query_breakdown.py

```diff
@@ -0,0 +1,267 @@
+            return None
+
+        # Get the node with the highest complexity score
+        _, child = pipeline_breaker_list.pop()
+
+        # Create a temp table for the partitioned node
+        temp_table_name = self._get_temp_table_name()
+        temp_table_plan = self._query_generator.resolve(
+            SnowflakeCreateTable(
+                [temp_table_name],
+                None,
+                SaveMode.ERROR_IF_EXISTS,
+                child,
+                table_type="temp",
+                creation_source=TableCreationSource.LARGE_QUERY_BREAKDOWN,
+            )
+        )
+
+        # Update the ancestors with the temp table selectable
+        self._replace_child_and_update_ancestors(parent_map, child, temp_table_name)
+
+        return temp_table_plan
+
+    def _is_node_valid_to_breakdown(self, score: int, node: LogicalPlan) -> bool:
+        """Method to check if a node is valid to breakdown based on complexity score and node type."""
+        return (
+            COMPLEXITY_SCORE_LOWER_BOUND < score < COMPLEXITY_SCORE_UPPER_BOUND
+        ) and self._is_node_pipeline_breaker(node)
+
+    def _is_node_pipeline_breaker(self, node: LogicalPlan) -> bool:
+        """Method to check if a node is a pipeline breaker based on the node type.
+
+        If the node contains a SnowflakePlan, we check its source plan recursively.
+        """
+        if isinstance(node, (Pivot, Unpivot, Sort, Aggregate)):
+            return True
+
+        if isinstance(node, Sample):
+            return node.row_count is not None
+
+        if isinstance(node, Union):
+            return not node.is_all
+
+        if isinstance(node, SelectStatement):
+            return node.order_by is not None
+
+        if isinstance(node, SetStatement):
+            return any(operand.operator == SET_UNION for operand in node.set_operands)
+
+        if isinstance(node, SnowflakePlan):
+            return node.source_plan is not None and self._is_node_pipeline_breaker(
+                node.source_plan
+            )
+
+        if isinstance(node, (SelectSnowflakePlan, SelectTableFunction)):
+            return self._is_node_pipeline_breaker(node.snowflake_plan)
+
+        return False
+
+    def _replace_child_and_update_ancestors(
+        self, parent_map: defaultdict, child: LogicalPlan, temp_table_name: str
+    ) -> None:
+        """This method replaces the child node with a temp table selectable, resets
+        the snowflake plan and cumulative complexity score for the ancestors, and
+        updates the ancestors with the correct snowflake query corresponding to the
+        new plan tree.
+        """
+        temp_table_node = SnowflakeTable(temp_table_name, session=self.session)
+        temp_table_selectable = self._query_generator.create_selectable_entity(
+            temp_table_node, analyzer=self._query_generator
+        )
+
+        # add drop table in post action and see if it propagates to the root
+        drop_table_query = Query(
```

Comment:
add a comment here that the temp table created right now is query scoped, will be dropped after the query executed